packages/mobile/src/store/offline-downloads/sagas/offlineQueueSagas/workers/downloadTrackWorker.ts

```diff
@@ -157,13 +160,36 @@ function* downloadTrackAudio(track: UserTrackMetadata) {
   const trackFilePath = getLocalAudioPath(track_id)
   const encodedTrackId = encodeHashId(track_id)
 
-  for (const creatorNodeEndpoint of creatorNodeEndpoints) {
-    const trackAudioUri = `${creatorNodeEndpoint}/tracks/stream/${encodedTrackId}`
-    const response = yield* call(downloadFile, trackAudioUri, trackFilePath)
-    const { status } = response.info()
-    if (status === 200) return
+  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+  const apiClient = yield* getContext('apiClient')
+  const getFeatureEnabled = yield* getContext('getFeatureEnabled')
+  const isGatedContentEnabled = yield* call(() => {
```

Comment:
Haven't seen this pattern before. The one I'm familiar with is:
```
  const isNewPodcastControlsEnabled = yield* call(
    getFeatureEnabled,
    FeatureFlags.PODCAST_CONTROL_UPDATES_ENABLED,
    FeatureFlags.PODCAST_CONTROL_UPDATES_ENABLED_FALLBACK
  )
```

Reply:
yeah I've seen both. it's that typescript was complaining because of `getFeatureEnabled` returning a Promise. Weird that the other pattern works too. Can dig or chat with a client folk in case im missing sthg

Reply:
Yeah this is odd, worth digging in to figure out why our usual pattern doesn't work. @sddioulde looking around I found a few places in the codebase where I think we're not awaiting this promise:

```
  const getFeatureEnabled = yield* getContext('getFeatureEnabled')
  const isGatedContentEnabled = getFeatureEnabled(
    FeatureFlags.GATED_CONTENT_ENABLED
  )
```

Worth a search for `getFeatureEnabled` in `**/src/**/sagas**`

Reply:
Yeah I think the usual pattern should be possible

Reply:
fixed